hal/common/retarget.cpp

```diff
@@ -149,13 +149,21 @@ static inline int openmode_to_posix(int openmode) {
     return posix;
 }
 
+extern "C" WEAK void mbed_sdk_init(void);
+extern "C" WEAK void mbed_sdk_init(void) {
+}
+
 extern "C" FILEHANDLE PREFIX(_open)(const char* name, int openmode) {
     #if defined(__MICROLIB) && (__ARMCC_VERSION>5030000)
     // Before version 5.03, we were using a patched version of microlib with proper names
     // This is the workaround that the microlib author suggested us
     static int n = 0;
+    static int mbed_sdk_inited = 0;
+    if (!mbed_sdk_inited) {
+        mbed_sdk_inited = 1;
+        mbed_sdk_init();
```

Comment:
@c1728p9  About being called twice, I think that various calls are under different #defines - the one in _open is only for uARM toolchain, the other calls are in case of GCC or ARM toolchain. But I may be wrong, which line to you suspect to be the 2nd call ?